store/helpers/pages/pages.js

```diff
@@ -77,16 +80,25 @@ class Pages extends Store {
           this.isLastPageLoaded =
             this.isLastPageLoaded || transformedData.length === 0
           resolve(transformedData)
-        },
-        (reason) => {
-          if (reason instanceof NotFoundError) {
-            if (this.pageNumber === 0) this.isFirstPageLoaded = true
-            this.isLastPageLoaded = true
-            this.pageNumber += 1
+        })
+        .catch((error) => {
+          // Storing the error to propagate messages to the UI
+          this.error = error
+
+          // Resetting pointers to prevent pagination working
+          this.isFirstPageLoaded = true
+          this.isLastPageLoaded = true
+
+          if (error instanceof NotFoundError) {
+            this.data = []
+            resolve()
+          } else if (error instanceof PaymentRequiredError) {
+            // Stripping data to 5 elements only if there are more
+            // These elements may be used as a preview
+            this.data = this.data.slice(0, 5)
```

Comment:
Btw. what if data provider doesn't have any data? Should we still show the placeholder? We are basically selling no data. My point is whether 402 take precedence over 404. I would rather suggest to have it the other way around if it's not that way already. It's more a question for @oacore/api.

Reply:
I added a trick around this into the store. I agree, 404 should be more important. @oacore/api plase consider the following error precedence for _protected_ endpoints:
1. `401 Unauthorized` if the user is anonymous **(highest priority)**
2. `403 Forbidden` if the user is logged in but trying to get a resource they do not have access to
3. `404 Not Found` if the user has access rights but there is no data to provide (empty object or collection is returned)
4. `402 Payment Required` if the collection is not empty but the user is a free customer and cannot get access to the resource **(lowest priority)**

Reply:
Using this we need to run the full endpoint even when it shouldn't be allowed.
I would argue that 402 should be returned any case with data or not, therefore I would swap (3) and (4). 
I like the connection of status codes with semantic but I am not completely bought on using 404 so strictly for all the empty results.

Reply:
> It could be that there is no data in that endpoint but we are still selling a valuable product and data from other endpoints!

Exactly! I do not want to display the message in the wrong place. If something is not sellable we must not annoy the user.

It's done with a workaround now because I do not even know the real precedence. I would like to avoid this trick but feel free to swap (3) and (4) if you find it painful or unreasonable to do it the proposed way.